mp/src/game/shared/momentum/fx_mom_shared.cpp

```diff
@@ -141,16 +144,12 @@ void FX_FireBullets(int iEntIndex, const Vector &vOrigin, const QAngle &vAngles,
 
 #ifdef GAME_DLL
     // Weapon sounds are server-only for PAS ability
+    static ConVarRef mom_paint_apply_sound("mom_paint_apply_sound");
 
-    static ConVarRef paintgun_shoot_sound("mom_paintgun_shoot_sound");
-
-    // Do an extra paintgun check here
-    const bool bPreventShootSound = iAmmoType == AMMO_TYPE_PAINT && !paintgun_shoot_sound.GetBool();
-
-    if (!bPreventShootSound)
+    // Do an extra paint check here
+    if (iAmmoType == AMMO_TYPE_PAINT && mom_paint_apply_sound.GetBool())
     {
-        const auto pWeaponScript = g_pWeaponDef->GetWeaponScript(hWeaponID);
-        FX_WeaponSound(iEntIndex, pWeaponScript->pKVWeaponSounds->GetString("single_shot"), vOrigin);
```

Comment:
I believe this logic was messed up. `bPreventShootSound` was negated in the check, but what it did was allow the rest of the weapons to shoot as well. With this change it seems like no weapons would make any sound.

You can pull out the check for paint type ammo to a bool and check if it's not prevented, and if it's paint use the hardcoded sound otherwise get the sound from the weaponsounds KV.

Reply:
woops, I had though this code was specific to the paintgun when looking over it. I tested with stickylauncher and that still makes a sound but not the hitscan weapons, why is that?

Reply:
Changed now to 
```
    if (iAmmoType != AMMO_TYPE_PAINT)
    {
        const auto pWeaponScript = g_pWeaponDef->GetWeaponScript(hWeaponID);
        FX_WeaponSound(iEntIndex, pWeaponScript->pKVWeaponSounds->GetString("single_shot"), vOrigin);
    }
    else if (mom_paint_apply_sound.GetBool())
    {
        FX_WeaponSound(iEntIndex, SND_PAINT_SHOT, vOrigin);
    }
```